### PR TITLE
fix 1660 incorrect warning message when define an osimage

### LINKED ISSUE #1660: incorrect warning message when define an osimage
I have an stanza file: 
```
[root@c712ems4 0809]# cat /tmp/ww.file
# <xCAT data object stanza file>

rh72openpower4DL1632:
    objtype=osimage
    imagetype=linux
    osarch=ppc64le
    osdistroname=rhels7.2-ppc64le
    osname=Linux
    osvers=rhels7.2
    otherpkgdir=/install/post/otherpkgs/rhels7.2/ppc64le/rh72diskless1620
    otherpkglist=/install/custom/gpfs/gpfs.otherpkgs.pkglist,/install/custom/netboot/rh/rh72openpower4diskless1630b.cuda.otherpkgs
    permission=755
    pkgdir=/install/rhels7.2/ppc64le_patch,/install/rhel7220160512
    pkglist=/install/custom/netboot/rh/rh72diskless1625.rhels7.ppc64le.pkglist
    postbootscripts=mountEMS_install.sh,syncfilesfromEMS.sh,restorecap
    postinstall=/install/custom/netboot/rh/rh72diskless1620.ppc64le.postinstall
    postscripts=setupntp,local.tune
    profile=compute
    provmethod=netboot
    rootimgdir=/install/netboot/rhels7.2/ppc64le_patch/rh72openpower4DL1632
```
```
 cat /tmp/ww.file |mkdef -z
Warning: The node name 'rh72openpower4DL1632' contains capital letters which may not be resolved correctly by the dns server.
1 object definitions have been created or modified.
```
The "Warning: The node name ...." confused user.

### PR DESCRIPTION
**Pull request by bybai**
fix #1660

### CODE REVIEW

**Comment by immarvin** [comment]
hi @bybai , currently xCAT does not restrict the usage of capital letters in osimage names.
[root@c910f04x30v02 initrd]# mkdef -t osimage -o AAA pkgdir=pp imagetype=linux
1 object definitions have been created or modified.

[root@c910f04x30v02 initrd]# lsdef -t osimage -o AAA
Object name: AAA
imagetype=linux
pkgdir=pp

so I do not think this PR is necessary.

File: xCAT-server/lib/xcat/plugins/DBobjectdefs.pm [hunk 1]
```diff
@@ -1544,7 +1544,7 @@ sub defmk
     } else {
         my $invalidnodename = ();
         foreach my $node (@::allobjnames) {
-            if (($node =~ /[A-Z]/) && ((!$::opt_t) || ($::opt_t eq "node"))) {
```
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/DBobjectdefs.pm]: the the -t is not specified, it will be node by default

**Comment by bybai** [comment]
Hi @immarvin Based on issue 1660, this fix is for "mkdef -z", mkdef -t -o do not have this issue.

**Comment by immarvin** [comment]
hi @bybai , with your PR, the warning message of  "cat <node obj stanza>|mkdef -z"  disappeared.

**Comment by bybai** [comment]
My unit test:
cat osimage |mkdef -z
cat node|mkdef -z
cat node|chdef -z
cat osimage|chdef -z
mkdef TT cons=ipmi groups=all
mkdef  -t node -o TT cons=ipmi groups=all
mkdef -t osimage -o RH7 imagetype=linux osarch=ppc64le
chdef TT cons=ipmi groups=all
chdef  -t node -o TT cons=ipmi groups=all
chdef -t osimage -o RH7 imagetype=linux osarch=ppc64le

**Comment by immarvin** [comment]
agree to merge

**Pull request merged**